File: container-with-most-water.cpp
class Solution {
    public int maxArea(int[] height) {
        int right = height.length-1;
        int left = 0;
        int res = 0;
        while(left<right){
            res = Math.max(res,(right-left)*(Math.min(height[right], height[left])));
            if(height[right]>height[left]) left ++;
            else right --;
        }
        return res;
    }
    
}

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        // start from both ends
        // shrink from the smaller end
        // until the 2 ends meet
        // keep a max 
        int result = 0;
        int left = 0;
        int right = height.length - 1;
        while(left<right){
            int area = Math.min(height[left], height[right]) * (right-left);
            result = Math.max(result, area);
            if(height[left]<height[right]){
                left++;
            }
            else{
                right--;
            }
        }

        return result;
    }
}